source/common/http/match_delegate/config.cc

```diff
@@ -317,6 +324,8 @@ FilterConfigPerRoute::createFilterMatchTree(
     Server::Configuration::ServerFactoryContext& server_context) {
   auto requirements =
       std::make_unique<envoy::extensions::filters::common::dependency::v3::MatchingRequirements>();
+  requirements->mutable_data_input_allow_list()->add_type_url(TypeUtil::descriptorFullNameToTypeUrl(
+      envoy::type::matcher::v3::HttpRequestHeaderMatchInput::descriptor()->full_name()));
```

Comment:
The intention of matching requirements is to limit to request header. But with the introduction of CEL matcher, we want to allow CEL matcher input type `xds::type::matcher::v3::HttpAttributesCelMatchInput`. So the code here actually rejects CEL matcher.

Technically `HttpAttributesCelMatchInput` doesn't limit to request header, but we want to get rid of request header restriction eventually.

So here are two options:
1. Keep this requirement and add `xds::type::matcher::v3::HttpAttributesCelMatchInput`
2. Let's just leave it empty requirement as before. We can revisit here later.

I personally lean towards 2.